### PR TITLE
[Safety Checker] Add Safety Checker Module

### PR DESCRIPTION
**Pull request by patrickvonplaten**
With @anton-l and @patil-suraj, we've made sure that the safety_checker works correctly see here: https://github.com/huggingface/diffusers/pull/219

The only thing we're not 100% sure about is whether this line https://github.com/huggingface/diffusers/blob/89e9521048067acacfdcbc2b985af8f6b155cfb6/src/diffusers/pipelines/stable_diffusion/safety_checker.py#L65 that defines the threshold is correct

### CODE REVIEW

File: scripts/txt2img.py [hunk 1]
```diff
@@ -220,7 +237,9 @@ def main():
     if opt.fixed_code:
         start_code = torch.randn([opt.n_samples, opt.C, opt.H // opt.f, opt.W // opt.f], device=device)
 
+    print("start code", start_code.abs().sum())
```
Comment (patrickvonplaten) [inline scripts/txt2img.py:240]:
```suggestion
```

File: scripts/txt2img.py [hunk 2]
```diff
@@ -220,7 +237,9 @@ def main():
...
     precision_scope = autocast if opt.precision=="autocast" else nullcontext
+    precision_scope = nullcontext
```
Comment (patrickvonplaten) [inline scripts/txt2img.py:242]:
```suggestion
```

File: scripts/txt2img.py [hunk 3]
```diff
@@ -269,7 +288,11 @@ def main():
                     Image.fromarray(grid.astype(np.uint8)).save(os.path.join(outpath, f'grid-{grid_count:04}.png'))
                     grid_count += 1
 
-                toc = time.time()
+                image = x_samples_ddim.cpu().permute(0, 2, 3, 1).numpy()
```
Comment (patrickvonplaten) [inline scripts/txt2img.py:291]:
```suggestion
                toc = time.time()
                image = x_samples_ddim.cpu().permute(0, 2, 3, 1).numpy()
```

**Review by rromb — COMMENTED** [review]
Looks good, we could consider making this optional

**Comment by patrickvonplaten** [comment]
Happy to adapt the way you prefer

**Comment by woctezuma** [comment]
> > Or just a simple 'comment this out to completely take safety off' might also be a good note to have.
> 
> yes, would be nice to be able to remove the safety checker. some of us are adults!

You can already remove the safety, it is just that the removal procedure is not explained in the documentation.

**Comment by aisensiy** [comment]
I saw the model come from https://huggingface.co/CompVis/stable-diffusion-safety-checker/blob/main/pytorch_model.bin and I am wondering where is this model come from. There is some nsfw detector in github but this one looks different (larger).

**Comment by didpublishing** [comment]
So far I've had no luck finding the right script to remove the nsfw feature. It's blocking images that are already safe. Very frustrating.

**Comment by woctezuma** [comment]
> So far I've had no luck finding the right script.

Maybe try [this notebook][notebook].

[notebook]: <https://github.com/woctezuma/stable-diffusion-colab>

File: scripts/txt2img.py (hunk 1 above)
Comment (abrahem79) [reply scripts/txt2img.py]: > ```suggestion
> 
> ```

**Pull request merged**